hack/make/.binary

```diff
@@ -78,6 +78,17 @@ source "${MAKEDIR}/.go-autogen"
 		GCFLAGS="all=-N -l"
 	fi
 
+	if [ "$(go env GOARCH)" = "arm" ] && [ "$(go env GOARM)" = "5" ]; then
+		# cross-compiling for arm/v5 fails on go1.22; a fix is included for this
+		# in go1.23 (https://github.com/golang/go/issues/65290), but for go1.22
+		# we can set the correct option manually.
+		export CCGO_CFLAGS=-Wno-atomic-alignment
```

Comment:
You misspelled `CGO_CFLAGS`, which is serendipitous: it doesn't clobber the `CGO_CFLAGS` variable set on line 25. But it means the flag doesn't actually take effect, either. Either use `+=` [or append the `-Wno-` flag  to line 25.](https://github.com/corhere/moby/commit/f94c328f2e711b92fe9065c3c73b833fec0f362a)

Reply:
> You misspelled CGO_CFLAGS

DERP! My computer was jerky during the zoom call, and didn't notice 😂 


> [or append the -Wno- flag to line 25.](https://github.com/corhere/moby/commit/f94c328f2e711b92fe9065c3c73b833fec0f362a)

Oh! I like that one. I clearly didn't look at the file in a wider scope. Yes, that looks a lot cleaner as we already have the different variations there 👍

Reply:
Oh! But I guess that branch is only for cross-compiling, so wouldn't apply to non-cross-compile 🤔 I can probably use the other approach; and I just spotted you did so as well in your branch in https://github.com/moby/moby/pull/48397